magento/pwa-studio
# Allow self signed certs for backend validation and only attach to scaffold sample

## Description

The validator was running against every back end, even your own custom backend which you most likely will have configured when developing your storefront.

If you used a local backend with some self signed cert, it fails to validate:

```
~/code/pwa-studio develop
λ node
Welcome to Node.js v14.15.1.
Type ".help" for more information.
> const fetch = require('node-fetch');
undefined
> fetch('https://latest.test')
Promise { <pending> }
> (node:64113) UnhandledPromiseRejectionWarning: FetchError: request to https://latest.test/ failed, reason: unable to verify the first certificate
    at ClientRequest.<anonymous> (/Users/stephen/code/pwa-studio/node_modules/node-fetch/lib/index.js:1444:11)
    at ClientRequest.emit (events.js:315:20)
    at ClientRequest.EventEmitter.emit (domain.js:486:12)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at TLSSocket.EventEmitter.emit (domain.js:486:12)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:64113) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:64113) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

One solution would be to allow self signed certs through the `https` agent.

## Related Issue





Closes [PWA-1182](https://jira.corp.magento.com/browse/PWA-1182).

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


In all cases, I have removed the `console.warn` that was displaying on every `yarn` command.

1. Check out PR.
2. Run `yarn`. It shouldn't say anything because the extension isn't installed in the core repo.
3. Using the PR, scaffold a new app: `DEBUG_PROJECT_CREATION=1 pwa-studio/packages/create-pwa/bin/create-pwa`.
4. in the new app dir, run `yarn build` and you should see the console warning.
5. In the new app, swap the backend url to an invalid url and run `yarn build`. The error should display with backends to try.

Also, navigate to the `validate-sample-backends` extension directory and run `yarn test` to ensure the tests pass. I don't think our test runner validates extensions.

Try this with 

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.


### Resolved issues:
1. [x] resolves magento/pwa-studio#2894: Allow self signed certs for backend validation
